cpu/mips_pic32_common/periph/gpio.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright(C) 2017 Imagination Technologies Limited and/or its
+ *              affiliated group companies.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+#include <assert.h>
+#include <stdint.h>
+#include "periph/gpio.h"
+#include "board.h"
+
+#define GPIO_PIN_NO(PIN)    (1 << ((PIN) & 0xf))
+#define GPIO_PORT(PIN)      ((PIN) >> 4)
+
+#define LATx(P)         (base_address[P][0x10/0x4])
+#define LATxCLR(P)      (base_address[P][0x14/0x4])
+#define LATxSET(P)      (base_address[P][0x18/0x4])
+#define LATxINV(P)      (base_address[P][0x1C/0x4])
+#define PORTx(P)        (base_address[P][0x0])
+#define ANSELxCLR(P)    (base_address[P][(-0x20 + 0x4)/0x4])
```

Comment:
Jenkins has already barfed on this. Yes it'll work but if MCHP produce a new part where the ANSEL regs are at a different offset to the GPIO regs this won't work. I think you'll have to declare an 'analogue_port_base_address' using ANSEL{A,B,C,D,E} defines from the vendor headers.

Otherwise looks good.